bitarray/_bitarray.c

```diff
@@ -190,6 +190,39 @@ bytereverse(bitarrayobject *self, Py_ssize_t start, Py_ssize_t n)
 #define UINT64_WORDS(bytes)  0
 #endif
 
+static void
+shift1(bitarrayobject *self, int insert)
+{
+    Py_ssize_t i, nwords = UINT64_WORDS(Py_SIZE(self));
+    /* position of lowest bit in byte */
+    Py_ssize_t bit0 = self->endian == ENDIAN_LITTLE ? 0 : 7;
```

Comment:
Wouldn't the `bytereverse()` right after this cancel it out?

Reply:
Endianness is always tricky.  Yes, line 201 does cancel this out, and this is why depending on the endianness the `bit0` (ie. the bit with the lowest index) is 0 (LE), or 7 (BE).  The same reason you have two cases in the `BITMASK` definition in `bitarray.h`.